### PR TITLE
Prevent older fetchers from triggering search

### PR DESCRIPTION
**Pull request by paulfalgout**
Shortcut Story ID: [sc-46054]

Fairly simple check to make sure the latest fetch is the only one that resolves.

Not certain how to test this successfully if we're not getting coverage.

### CODE REVIEW

**Review by nmajor25 — APPROVED** [review]

**Comment by nmajor25** [comment]
Just noticed the coverage miss for the `if (this.fetcher !== fetcher) return;` line.

**Comment by paulfalgout** [comment]
I figured out a coverage test

File: test/integration/services/patient-quick-search.js
Comment (nmajor25) [inline test/integration/services/patient-quick-search.js:1]: This is purely for coverage, right? Like if we remove the `entities-service/entities/patient-search-results.js` file changes, this test will still pass.

Comment (nmajor25) [reply test/integration/services/patient-quick-search.js:1]: Well I guess the `if (this.fetcher !== fetcher) return;` line still isn't covered 🤔

Comment (paulfalgout) [reply test/integration/services/patient-quick-search.js:1]: yep it does.. although.. I didn't look at the `req.reply` API, but it's odd to be that the data wasn't nested in `body`....

Comment (paulfalgout) [reply test/integration/services/patient-quick-search.js:1]: it's covered isn't it?

Comment (paulfalgout) [reply test/integration/services/patient-quick-search.js:1]: hmm lemme run it again
it works locally.

Comment (nmajor25) [reply test/integration/services/patient-quick-search.js:1]: Ya it's probably just something weird with the coveralls report...

Comment (paulfalgout) [reply test/integration/services/patient-quick-search.js:1]: ok.. I think I got it working on CI by slowing down the test a hair.. ironically that flow.owner flake on coverage happened.. so running it.. _again_

**Pull request merged**